__tests__/app.test.js

```diff
@@ -0,0 +1,325 @@
+          return Promise.all([result, expectedResult]).then((response) => {
+            expect(response[0]).toEqual(response[1]);
+            expect(typeof response[0]).not.toBe("string");
+          });
+        });
+    });
+  });
+});
+
+describe("/api/topics", () => {
+  describe("GET", () => {
+    test("returns a 200 status on successful request", () => {
+      return request(app).get("/api/topics").expect(200);
+    });
+    test("returns an object with a key of topics that contains an array of objects", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((response) => {
+          const topics = response.body.topics;
+          expect(Array.isArray(topics)).toBe(true);
+          expect(typeof topics[0]).toBe("object");
+        });
+    });
+    test("each object in topics array has 2 keys, a slug and a description", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((response) => {
+          const topics = response.body.topics;
+          expect(topics.length).toBe(3);
+          topics.forEach((topic) => {
+            expect(Object.keys(topic)).toEqual(["slug", "description"]);
+          });
+        });
+    });
+  });
+});
+
+describe("/api/articles/:id", () => {
+  describe("GET", () => {
+    test("returns a 200 status and a body with the correct keys", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then((response) => {
+          const article = response.body.article;
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+        });
+    });
+    test("Returns a 400 bad request error when given an :id parameter which is not valid", () => {
+      return request(app)
+        .get("/api/articles/elephant")
+        .expect(400)
+        .then((response) => expect(response.body.msg).toBe("Bad Request"));
+    });
+    test("Returns a 404 not found error when given a valid Id that does not exist", () => {
+      return request(app)
+        .get("/api/articles/99999")
+        .expect(404)
+        .then((response) => expect(response.body.msg).toBe("Not Found"));
+    });
+  });
+  describe("PATCH", () => {
+    test("Returns 200 status and the correct update article when sent a body containing votes to update", () => {
+      const patchUpdate = { inc_votes: 1 };
+      return request(app)
+        .patch("/api/articles/1")
+        .send(patchUpdate)
+        .expect(200)
+        .then((response) => {
+          const article = response.body.article;
+          expect(article.votes).toBe(101);
```

Comment:
We have already tested for the shape of an article. It is enough in this test to just check that the votes column has been updated.